xla/service/cpu/onednn_matmul.cc

```diff
@@ -100,45 +101,61 @@ ABSL_ATTRIBUTE_NO_SANITIZE_MEMORY void __xla_cpu_runtime_OneDnnMatMul(
   auto bias_mem = memory(nullptr);
   auto result_mem = memory(result_md, cpu_engine, result_minfo.Data());
 
-  bool bias_fusion = (!matmul_config.fused_ops().empty() &&
-                      matmul_config.fused_ops(0) == OneDnnMatMulConfig::BIAS);
-
-  auto matmul_pd =
-      matmul::primitive_desc(cpu_engine, lhs_md, rhs_md, result_md);
-
-  if (bias_fusion) {
-    auto bias_mem_desc = [](MemrefInfo base, dnnl::memory::data_type dtype) {
-      std::vector<int64_t> dims(base.GetRank(), 1);
-      std::vector<int64_t> strides(base.GetRank(), 1);
-      dims.at(base.GetRank() - 1) = base.GetChannels();
-      return memory::desc{dims, dtype, strides};
-    };
-    MemrefInfo bias_minfo(args[arg_indx++]);
-    bias_md = bias_minfo.GetOneDnnMemDesc();
-
-    // extend bias rank to match result rank
-    auto missed_rank = result_md.get_ndims() - bias_md.get_ndims();
-    XLA_LIGHTWEIGHT_CHECK(missed_rank >= 0);
-    if (missed_rank > 0) {
-      auto bias_dims = bias_md.get_dims();
-      bias_dims.insert(bias_dims.begin(), missed_rank, 1);
-      bias_md = bias_md.reshape(bias_dims);
+  // Currently, GELU/ReLU only fusion is supported.
+  dnnl::post_ops post_ops;
+  for (auto& fused_op : matmul_config.fused_ops()) {
+    switch (fused_op) {
+      case OneDnnMatMulConfig::RELU:
+        post_ops.append_eltwise(dnnl::algorithm::eltwise_relu, 0.f, 0.f);
+        break;
+      case OneDnnMatMulConfig::TANH:
+        post_ops.append_eltwise(dnnl::algorithm::eltwise_tanh, 0.f, 0.f);
+        break;
+      case OneDnnMatMulConfig::GELU_TANH:
+        post_ops.append_eltwise(dnnl::algorithm::eltwise_gelu_tanh, 0.f, 0.f);
+        break;
+      case OneDnnMatMulConfig::GELU_ERF:
+        post_ops.append_eltwise(dnnl::algorithm::eltwise_gelu_erf, 0.f, 0.f);
+        break;
+      case OneDnnMatMulConfig::BIAS: {
+        MemrefInfo bias_minfo(args[arg_indx++]);
+        bias_md = bias_minfo.GetOneDnnMemDesc();
+
+        // Extend bias rank to match result rank.
+        auto missed_rank = result_md.get_ndims() - bias_md.get_ndims();
+        XLA_LIGHTWEIGHT_CHECK(missed_rank >= 0);
+        if (missed_rank > 0) {
+          auto bias_dims = bias_md.get_dims();
+          bias_dims.insert(bias_dims.begin(), missed_rank, 1);
+          bias_md = bias_md.reshape(bias_dims);
+        }
+        bias_mem = memory(bias_md, cpu_engine, bias_minfo.Data());
+      } break;
+      default:
+        LOG(FATAL) << __FILE__ << ":" << __LINE__
+                   << " Attempt to call OneDNN MatMul runtime library with "
+                      "unsupported post op."
+                   << std::endl;
+        std::abort();
```

Comment:
Sorry I didn't catch this part yesterday. LOG already prints a new line at the end, and LOG(FATAL) terminates the program after printing, so we don't need `std::abort()`.

Reply:
Oh, yes, I missed it as well :) .
Fixed.